### DISCUSSION TITLE
2.1.7 build fail on linux

### DISCUSSION THREAD

**Discussion by chunlei-wu** [microsoft/msquic#3414] [Q&A] [answered]
When I try to build the 2.1.7 on linux, it failure with the following information, and building pass with 2.1.5.
platform_posix.c:40:10: fatal error: numa.h: no such file or directory

who can give me some suggetion.

**Comment by ManickaP** [comment]
Install libnuma-dev.

@nibanks I think this is worth fixing, we hit the same issue in our stress tests https://github.com/dotnet/runtime/issues/81706, @wfurt relevant comment: https://github.com/dotnet/runtime/issues/81706#issuecomment-1419625199

**Reply by nibanks** [reply]
@manickap we have logic that is supposed to dynamically figure out if the dependency is installed and only enable it when present:
`cmakelists.txt`
```cmake
    find_library(NUMA NAMES NUMA libnuma.so.1)
    if (NUMA)
        message(STATUS "Found libnuma: ${NUMA}")
        list(APPEND QUIC_COMMON_DEFINES CXPLAT_NUMA_AWARE)
    else()
        message(STATUS "libnuma not found. If build fails, install libnuma")
    endif()
```
`platform_posix.c`
```c
#ifdef __aarch64__
#undef CXPLAT_NUMA_AWARE
#endif

#ifdef CXPLAT_NUMA_AWARE
#include <numa.h>
uint32_t CxPlatNumaNodeCount;
cpu_set_t* CxPlatNumaNodeMasks;
#endif // CXPLAT_NUMA_AWARE
```

But _sometimes_ this still doesn't work. I'd be happy to take a PR to fix it if you know how.

**Reply by nibanks** [reply]
FYI, my attempt at improving the situation: https://github.com/microsoft/msquic/pull/3415

**Comment by nibanks** [comment, marked as answer]
Are you 100% sure you're building 2.1.7 and not main? I just double checked and AFAICT we did not backport the libnuma stuff to 2.1. Yes, main has a new dependency there, but it shouldn't be in 2.1.

**Reply by nibanks** [reply]
I'm quite sure you were building main. Here is the release/2.1 file:

https://github.com/microsoft/msquic/blob/release/2.1/src/platform/platform_posix.c#L40

Here is main:

https://github.com/microsoft/msquic/blob/main/src/platform/platform_posix.c#L40

**Reply by chunlei-wu** [reply]
Yep, you are right. it's mistake, 2.1.7 build pass, Sorry for it.
Many thanks.
